param.go

```diff
@@ -299,7 +299,13 @@ func (ps paramSingle) Build(c containerStore, decorating bool) (reflect.Value, e
 	}
 
 	for _, n := range providers {
-		err := n.Call(c)
+		var err error
+		if n.Exported() {
+			err = n.Call(n.OrigScope())
+		} else {
+			err = n.Call(c)
+		}
```

Comment:
Could this just be `n.Call(n.OrigScope())` unconditionally? Unless I'm misunderstanding, either `Exported() == true` or `OrigScope() == c`.


```suggestion
		err := n.Call(n.OrigScope())
```

As an experiment, I tried this in a local checkout and only this test started failing:
https://github.com/uber-go/dig/blob/3a148a28d5453a4527f72b23dc44366ac39ca3db/scope_test.go#L351

Upon inspection, looks like the failure is correct? The `child.Provide`s are not exported so the parent should not see them. If I add `dig.Export(true)` to them, the test passes again.

```diff
diff --git a/scope_test.go b/scope_test.go
index cd7ec10..3710f75 100644
--- a/scope_test.go
+++ b/scope_test.go
@@ -371,9 +371,9 @@ func TestScopeValueGroups(t *testing.T) {
 		})
 
 		child := root.Scope("child")
-		child.RequireProvide(func() string { return "a" }, dig.Group("foo"))
-		child.RequireProvide(func() string { return "b" }, dig.Group("foo"))
-		child.RequireProvide(func() string { return "c" }, dig.Group("foo"))
+		child.RequireProvide(func() string { return "a" }, dig.Group("foo"), dig.Export(true))
+		child.RequireProvide(func() string { return "b" }, dig.Group("foo"), dig.Export(true))
+		child.RequireProvide(func() string { return "c" }, dig.Group("foo"), dig.Export(true))
 
 		// Invocation in child should see values provided to the child,
 		// even though the constructor we're invoking is provided in
```